src/Core/Framework/DataAbstractionLayer/VersionManager.php

```diff
@@ -772,25 +772,79 @@ private function buildWrites(VersionCommitCollection $commits): array
             $writes['delete']['version_commit'][] = ['id' => $commit->getId()];
         }
 
+        $this->cleanupSlotsReferencingDeletedBlocks($writes);
```

Comment:
Could we use an Event to Trigger this logic? Because in my opinion it should be in the CMS domain because it's very cms specific.